### ISSUE TITLE
Limit in-memory flows to a specific number

### ISSUE THREAD

**Issue by yannickbt64** [mitmproxy/mitmproxy#6398]
#### Problem Description

Mitmproxy memory usage will grow as more flows are added. Large downloads can cause issues.

#### Proposal
I'd like to have an option to limit the number of flows in-memory. Having a limit of e.g. 100 flows and progressively deleting the oldest one as a new one takes its place just like a FIFO.

#### Alternatives
Unless I am mistaken I'm not sure its possible right now to limit flows saved into memory.

#### Additional context
This is mostly when using the proxy for a long time.

**Comment by Prinzhorn** [comment]
This is a duplicate of https://github.com/mitmproxy/mitmproxy/issues/4129 and the comment there should still hold true.

**Comment by Prinzhorn** [comment]
> Large downloads can cause issues.

FYI you can use the ` stream_large_bodies` to not store any bodies larger than X, e.g. `3m`

**Comment by yannickbt64** [comment]
I'm partially sure (unless I don't fully understand) the answer and the forwarded comment satisfies my need.

I'm using mitmproxy for automated TLS/HTTPS interception and standard HTTP interception, to automatically reformat some requests, as well as also act, within the mitmproxy plugin, as a temporary HTTP(S) client (that is, act as a client to the upstream server) using pycurl to fetch additional data at the response and thus send a modified response back to the client of mitmproxy.

Once the client (the client of mitmproxy, the one who initiated the request I then modified seamlessly) receives the response, the flow is only nice if it is small, otherwise mitmproxy becomes slow uselessly.
On top of that, many many times, my mitmproxy plugin doesn't intervene.

Once that is done, if the flow is too big, e.g. 1k, I want to simply delete it from the memory.

Let me ask : Once a flow has been modified/sent to the requester (the client), is there a way to command its deletion? Can I do something, within e.g. the response hook, something like:
if not_a_request_i_must_intervene_on(flow.request):
    flow.response.send() # Send to the downstream client
    flow.mark_as_unlogged() # Don't log that, its useless for my needs

Ofc this is pure pseudocode but you can see the logic...

**Comment by Prinzhorn** [comment]
Please post this as a new discussion
